demisto_sdk/commands/create_id_set/tests/update_id_set_test.py

```diff
@@ -147,7 +149,122 @@ def test_create_id_set_on_specific_empty_pack(self, repo):
+        'author': 'Someone'
+    }
+
+    TEST_PACK = [
+        (METADATA_WITH_XSOAR_SUPPORT, 'Cortex XSOAR', 'certified'),
+        (METADATA_WITH_PARTNER_SUPPORT, 'Some Partner', 'certified'),
+        (METADATA_WITH_COMMUNITY_SUPPORT, 'Someone', ''),
+    ]
+
+    @staticmethod
+    @pytest.mark.parametrize('metadata_file_content, author, certification', TEST_PACK)
+    def test_process_metadata(repo, metadata_file_content, author, certification):
+        """
+        Given
+            - A pack_metadata file for Pack1
+        When
+            - parsing pack metadata files
+        Then
+            - parsing all the data from file successfully
+        """
+        pack = repo.create_pack("Pack1")
+        pack.pack_metadata.write_json(metadata_file_content)
+
+        res = get_pack_metadata_data(pack.pack_metadata.path)
+        result = res.get(list(res.keys())[0])
+
+        assert 'name' in result.keys()
+        assert result.get('name') == 'Pack1'
+        assert result.get('current_version') == '1.0.0'
+        assert result.get('author') == author
+        assert result.get('certification') == certification
+
+    @staticmethod
+    def test_get_metadata_bad_path(repo, mocker):
+        """
+        Given
+            - A bad path for a pack.
+        When
+            - Parsing pack metadata files.
+        Then
+            - Handle the exceptions gracefully.
+        """
+        import demisto_sdk.commands.common.update_id_set as uid
+        mocker.patch.object(uid, 'get_json', return_value={'name': 'Pack1'})
+
+        res = get_pack_metadata_data('Pack1')
+        result = res.get(list(res.keys())[0])
+
+        assert 'name' in result.keys()
+        assert not result.get('id')
+
+    @staticmethod
+    @pytest.mark.parametrize('print_logs', [True, False])
+    def test_process_packs_success(capsys, mocker, print_logs):
+        """
+        Given
+            - A pack metadata file path.
+            - Whether to print information to log.
+        When
+            - Parsing pack metadata files.
+        Then
+            - Verify output to logs.
+        """
+        import demisto_sdk.commands.common.update_id_set as uid
+        mocker.patch.object(uid, 'get_pack_metadata_data', return_value={'name': 'Pack'})
+
+        res = process_pack_metadata('Pack_Path', print_logs)
+        captured = capsys.readouterr()
+
+        assert res == [{'name': 'Pack'}]
+        assert ('adding Pack_Path to id_set' in captured.out) == print_logs
+
+    @staticmethod
+    def test_process_packs_exception_thrown(capsys, mocker):
+        """
+        Given
+            - A pack metadata file path.
+        When
+            - Parsing pack metadata files.
+        Then
+            - Handle the exceptions gracefully.
+        """
+        import demisto_sdk.commands.common.update_id_set as uid
+        # mocking some exception
+        mocker.patch.object(uid, 'get_pack_metadata_data', side_effect=lambda x: 1 / 0)
```

Comment:
If you don't know which exception would be raised here, it's better to raise a general exception, instead of raising a specific not relevant one.

Reply:
done